src/server.c

```diff
@@ -6810,7 +6810,7 @@ serverTestProc *getTestProcByName(const char *name) {
 }
 #endif
 
-int main(int argc, char **argv) {
+__attribute__((weak)) int main(int argc, char **argv) {
```

Comment:
```suggestion
/* Main is marked as weak so that unit tests can use their own main function. */
__attribute__((weak)) int main(int argc, char **argv) {
```